### PR TITLE
CO-2880 Email aliases interaction resume 

### PR DESCRIPTION
**Pull request by quentingigon**

### CODE REVIEW

File: crm_compassion/models/interaction_resume.py [hunk 1]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
-                WHERE m.subject IS NOT NULL
-                AND m.message_type = 'email'
-                AND (p.contact_id = %s OR p.id = %s)
-                )
-                """, [partner_id] * 8)
-        for row in self.env.cr.dictfetchall():
-            # ensure that "Example <ex@exmaple.com>" is correctly printed
-            row['email'] = html_sanitize(row['email'])
-            self.create(row)
+        original_partner = self.env['res.partner'].search([
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]:
```suggestion
        original_partner = self.env['res.partner'].browse(partner_id)
```

File: crm_compassion/models/interaction_resume.py [hunk 2]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+            ('id', '=', partner_id)
+        ])
+        email_address = original_partner.email
+        partners_with_same_email_ids = self.env['res.partner'].search([
+            ('email', '=', email_address)
+        ]).mapped("id")
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]:
```suggestion
        ]).ids
```

File: crm_compassion/models/interaction_resume.py [hunk 3]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
-        for row in self.env.cr.dictfetchall():
-            # ensure that "Example <ex@exmaple.com>" is correctly printed
-            row['email'] = html_sanitize(row['email'])
-            self.create(row)
+        original_partner = self.env['res.partner'].search([
+            ('id', '=', partner_id)
+        ])
+        email_address = original_partner.email
+        partners_with_same_email_ids = self.env['res.partner'].search([
+            ('email', '=', email_address)
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py:65]: We don't want to group the partners that don't have email address together!
```suggestion
            ('email', '=', email_address),
            ('email', '!=', False)
```

File: crm_compassion/models/interaction_resume.py [hunk 4]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+        ]).mapped("id")
+
+        for partner_id in partners_with_same_email_ids:
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]: I think you don't need to loop in the partner_ids, you can just include the partner ids list inside the SQL query to make it more efficient. (I think one SQL query is more efficient than executing several).

File: crm_compassion/models/interaction_resume.py [hunk 5]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+            self.search([('partner_id', '=', partner_id)]).unlink()
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]:
```suggestion
        self.search([('partner_id', 'in', partners_with_same_email_ids)]).unlink()
```

File: crm_compassion/models/interaction_resume.py [hunk 6]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+                        m.id as message_id,
+                        0 as paper_id,
+                        NULL as tracking_status
+                        FROM "mail_message" as m
+                        JOIN res_partner p ON m.author_id = p.id
+                        WHERE m.subject IS NOT NULL
+                        AND m.message_type = 'email'
+                        AND (p.contact_id in ({}) OR p.id in ({}))
+                        )
+                        """.format(partner_id, partner_id, partner_id, partner_id,
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]: Using format with an SQL query is a security issue, you should use the %s like it was done before and pass the parameters directly to the execute function.

File: crm_compassion/models/interaction_resume.py [hunk 7]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+                        pcj.id as paper_id,
+                        NULL as tracking_status
+                        FROM "partner_communication_job" as pcj
+                        JOIN res_partner p ON pcj.partner_id = p.id
+                        FULL OUTER JOIN partner_communication_config c
+                            ON pcj.config_id = c.id
+                            AND c.name != 'Default communication'
+                        WHERE pcj.state = 'done'
+                        AND pcj.send_mode = 'physical'
+                        AND (p.contact_id in ({}) OR p.id in ({}))
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]: ==> These parameters should be replaced by `original_partner_id`. We don't want to mix the other partners for physical communications.
```suggestion
                        AND (p.contact_id = %s OR p.id = %s)
```

File: crm_compassion/models/interaction_resume.py [hunk 8]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+                            END
+                        AS direction,
+                        crmpc.id as phone_id,
+                        0 as email_id,
+                        0 as message_id,
+                        0 as paper_id,
+                        NULL as tracking_status
+                        FROM "crm_phonecall" as crmpc
+                        JOIN res_partner p ON crmpc.partner_id = p.id
+                        WHERE (p.contact_id in ({}) OR p.id in ({}))
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]: ==> These parameters should be replaced by `original_partner_id`. We don't want to mix the other partners for physical communications.
```suggestion
                        WHERE (p.contact_id = %s OR p.id = %s)
```

File: crm_compassion/models/interaction_resume.py [hunk 9]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+                        ON rel.mail_mail_id = mail.id
+                        FULL OUTER JOIN mail_tracking_email mt ON mail.id = mt.mail_id
+                        JOIN res_partner p ON rel.res_partner_id = p.id
+                        FULL OUTER JOIN partner_communication_job job
+                            ON job.email_id = mail.id
+                        FULL OUTER JOIN partner_communication_config config
+                            ON job.config_id = config.id
+                            AND config.name != 'Default communication'
+                        WHERE mail.state = ANY (ARRAY ['sent', 'received'])
+                        AND (p.contact_id in ({}) OR p.id in ({}))
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]: These parameters will be replaced by `partners_with_same_email_ids`
```suggestion
                        AND (p.contact_id = ANY(%s) OR p.id = ANY(%s))
```

File: crm_compassion/models/interaction_resume.py [hunk 10]
```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
...
+                        0 as phone_id,
+                        0 as email_id,
+                        m.id as message_id,
+                        0 as paper_id,
+                        NULL as tracking_status
+                        FROM "mail_message" as m
+                        JOIN res_partner p ON m.author_id = p.id
+                        WHERE m.subject IS NOT NULL
+                        AND m.message_type = 'email'
+                        AND (p.contact_id in ({}) OR p.id in ({}))
```
Comment (ecino) [inline crm_compassion/models/interaction_resume.py]: These parameters will be replaced by `partners_with_same_email_ids`
```suggestion
                        AND (p.contact_id = ANY(%s) OR p.id = ANY(%s))
```

**Review by ecino — CHANGES_REQUESTED** [review]

File: crm_compassion/models/interaction_resume.py (hunk 4 above)
Comment (quentingigon) [reply crm_compassion/models/interaction_resume.py]: That's what I was trying to do at first but it was not working, I can try again

Comment (quentingigon) [reply crm_compassion/models/interaction_resume.py]: It's done now

File: crm_compassion/models/res_partner.py [hunk 11]
```diff
@@ -77,13 +77,16 @@ def open_interaction(self):
         """
         self.ensure_one()
         self.env['interaction.resume'].populate_resume(self.id)
+        partners_with_same_email_ids = self.env['res.partner'].search([
+            ('email', '=', self.email)
+        ]).mapped("id")
```
Comment (ecino) [inline crm_compassion/models/res_partner.py]:
```suggestion
        ]).ids
```

File: crm_compassion/models/interaction_resume.py (hunk 1 above)
Comment (ecino) [reply crm_compassion/models/interaction_resume.py]: You switched back to an older version and need to fix that again.

File: crm_compassion/models/interaction_resume.py (hunk 3 above)
Comment (ecino) [reply crm_compassion/models/interaction_resume.py:65]: Please restore this fix.

File: crm_compassion/models/res_partner.py [hunk 12]
```diff
@@ -77,13 +77,16 @@ def open_interaction(self):
         """
         self.ensure_one()
         self.env['interaction.resume'].populate_resume(self.id)
+        partners_with_same_email_ids = self.env['res.partner'].search([
+            ('email', '=', self.email)
```
Comment (ecino) [inline crm_compassion/models/res_partner.py:82]: Please add the filter `('email', '!=', False)` here as well

File: crm_compassion/models/res_partner.py [hunk 13]
```diff
@@ -77,13 +77,16 @@ def open_interaction(self):
...
+        ]).ids
         return {
             'name': _('Interaction resume'),
             'type': 'ir.actions.act_window',
             'res_model': 'interaction.resume',
             'view_type': 'form',
             'view_mode': 'tree,form',
-            'domain': [('partner_id', '=', self.id)],
+            'domain': [('partner_id', 'in', partners_with_same_email_ids)],
```
Comment (ecino) [inline crm_compassion/models/res_partner.py]:
```suggestion
            'domain': [('partner_id', 'in', self.ids + partners_with_same_email_ids)],
```

**Review by ecino — CHANGES_REQUESTED** [review]

**Review by ecino — APPROVED** [review]

**Pull request merged**